gef.py

```diff
@@ -11301,112 +11090,503 @@ def screen_setup(self):
+        """Write `buffer` at address `address`."""
+        return gdb.selected_inferior().write_memory(address, buffer, length)
+
+    def read(self, addr: int, length: int = 0x10) -> bytes:
+        """Return a `length` long byte array with the copy of the process memory at `addr`."""
+        return gdb.selected_inferior().read_memory(addr, length).tobytes()
+
+    def read_integer(self, addr: int) -> int:
+        """Return an integer read from memory."""
+        sz = gef.arch.ptrsize
+        mem = self.read(addr, sz)
+        unpack = u32 if sz == 4 else u64
+        return unpack(mem)
+
+    def read_cstring(self, address: int, max_length: int = GEF_MAX_STRING_LENGTH, encoding: Optional[str] = None) -> str:
+        """Return a C-string read from memory."""
+        encoding = encoding or "unicode-escape"
+        length = min(address | (DEFAULT_PAGE_SIZE-1), max_length+1)
+
+        try:
+            res_bytes = bytes(self.read(address, length))
+        except gdb.error:
+            err(f"Can't read memory at '{address}'")
+            return ""
+        try:
+            with warnings.catch_warnings():
+                # ignore DeprecationWarnings (see #735)
+                warnings.simplefilter("ignore")
+                res = res_bytes.decode(encoding, "strict")
+        except UnicodeDecodeError:
+            # latin-1 as fallback due to its single-byte to glyph mapping
+            res = res_bytes.decode("latin-1", "replace")
+
+        res = res.split("\x00", 1)[0]
+        ustr = res.replace("\n", "\\n").replace("\r", "\\r").replace("\t", "\\t")
+        if max_length and len(res) > max_length:
+            return f"{ustr[:max_length]}[...]"
+        return ustr
+
+    def read_ascii_string(self, address: int) -> Optional[str]:
+        """Read an ASCII string from memory"""
+        cstr = self.read_cstring(address)
+        if isinstance(cstr, str) and cstr and all([x in string.printable for x in cstr]):
+            return cstr
+        return None
+
+    @property
+    def maps(self):
+        if not self.__maps:
+            self.__maps = self.__parse_maps()
+        return self.__maps
+
+    def __parse_maps(self):
+        """Return the mapped memory sections"""
         try:
-            pyenv = which("pyenv")
-            PYENV_ROOT = gef_pystring(subprocess.check_output([pyenv, "root"]).strip())
-            PYENV_VERSION = gef_pystring(subprocess.check_output([pyenv, "version-name"]).strip())
-            site_packages_dir = os.path.join(PYENV_ROOT, "versions", PYENV_VERSION, "lib",
-                                             "python{}".format(PYENV_VERSION[:3]), "site-packages")
-            site.addsitedir(site_packages_dir)
+            return list(self.__parse_procfs_maps())
         except FileNotFoundError:
-            pass
+            return list(self.__parse_gdb_info_sections())
+
+    def __parse_procfs_maps(self):
```

Comment:
```suggestion
    def __parse_procfs_maps(self) -> Generator[Section]:
```

Reply:
@daniellimws this will cause an error. it has to be `Generator[Section, None, None]`

Reply:
Ah right, my bad. I think we can do `Iterator[Section]` instead, cleaner.